samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -74,7 +74,7 @@ public GroupByContainerCount(int containerCount) {
     // Convert to a Set of ContainerModel
     Set<ContainerModel> containerModels = new HashSet<>();
     for (int i = 0; i < containerCount; i++) {
-      containerModels.add(new ContainerModel(i, taskGroups[i]));
+      containerModels.add(new ContainerModel(String.valueOf(i), taskGroups[i]));
```

Comment:
A better long term approach is to completely overhaul these groupers to have no assumptions about the strings and treat them as arbitrary tokens.